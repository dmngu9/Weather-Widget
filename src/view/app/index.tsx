export { default as App } from './app-dumb';
