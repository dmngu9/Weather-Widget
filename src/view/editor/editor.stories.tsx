import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import Editor from './editor-dumb';

storiesOf('Editor', module).add('basic', () => <Editor unit="metric" wind="on" onChange={action('onchange')} />);
