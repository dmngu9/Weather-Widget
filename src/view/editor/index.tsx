export { default as Editor } from './editor-dumb';
export { FieldText, RadioButton } from './styles';
