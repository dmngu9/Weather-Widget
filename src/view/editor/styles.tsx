import styled from 'styled-components';

import * as fonts from 'utils/fonts';

export const Form = styled.form`
    width: 360px;
    padding: 24px;
`;

export const FieldSet = styled.fieldset`
    margin: 0 0 16px;
    padding: 0;
    border: none;
`;

export const Legend = styled.legend`
    padding: 0;
    margin-bottom: 8px;
    font-size: ${fonts.medium}px;
`;

export const FieldText = styled.input`
    height: 36px;
    width: 100%;
    border-radius: 4px;
    border: solid 1px #ccc;
    padding: 16px;
    box-sizing: border-box;
    outline: none;
    font-size: ${fonts.small}px;

    &:focus {
        border-color: dodgerblue;
        box-shadow: 0 0 8px 0 dodgerblue;
    }

    &::placeholder {
        font-size: ${fonts.small}px;
    }
`;

export const RadioLabel = styled.label`
    display: inline-block;
    cursor: pointer;
    font-size: ${fonts.small}px;
    user-select: none;
    padding-left: 24px;
    position: relative;

    &:hover > span {
        background-color: #ccc;
    }
`;

export const RadioButton = styled.input`
    display: none;

    &:checked + span {
        background-color: dodgerblue;
    }

    &:checked + span::after {
        content: '';
        height: 8px;
        width: 8px;
        background-color: white;
        border-radius: 50%;
        position: absolute;
        left: 50%;
        top: 50%;
        transform: translate(-50%, -50%);
    }
`;

export const Circle = styled.span`
    display: inline-block;
    height: 16px;
    width: 16px;
    background-color: #fff;
    position: absolute;
    left: 0;
    top: 0;
    border-radius: 50%;
`;

export const RadioGroup = styled.div`
    display: grid;
    grid-template-columns: 1fr 1fr;
`;
