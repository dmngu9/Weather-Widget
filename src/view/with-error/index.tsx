export { default as withError } from './with-error';
