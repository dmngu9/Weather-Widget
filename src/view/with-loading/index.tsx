export { default as withLoading } from './with-loading';
