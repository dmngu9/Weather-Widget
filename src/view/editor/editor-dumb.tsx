import * as React from 'react';

import { Form, Legend, FieldSet, FieldText, RadioLabel, RadioButton, Circle, RadioGroup } from './styles';

interface Props {
    unit: 'metric' | 'imperial';
    wind: 'on' | 'off';
    className?: string;
    onChange: (e: React.SyntheticEvent<HTMLInputElement>) => void;
}

const EditorDumb: React.StatelessComponent<Props> = ({ unit, wind, className, onChange }) => (
    <Form className={className}>
        <FieldSet>
            <Legend>Title</Legend>
            <FieldText type="text" name="title" maxLength={50} placeholder="Title of widget" onChange={onChange} />
        </FieldSet>
        <FieldSet>
            <Legend>Unit</Legend>
            <RadioGroup>
                <RadioLabel>
                    Metric (&#xb0;C, m/s)
                    <RadioButton
                        type="radio"
                        name="unit"
                        value="metric"
                        defaultChecked={unit === 'metric'}
                        onClick={onChange}
                    />
                    <Circle />
                </RadioLabel>
                <RadioLabel>
                    Imperial (&#xb0;F, mph)
                    <RadioButton
                        type="radio"
                        name="unit"
                        value="imperial"
                        defaultChecked={unit === 'imperial'}
                        onClick={onChange}
                    />
                    <Circle />
                </RadioLabel>
            </RadioGroup>
        </FieldSet>
        <FieldSet>
            <Legend>Wind</Legend>
            <RadioGroup>
                <RadioLabel>
                    On
                    <RadioButton
                        type="radio"
                        name="wind"
                        value="on"
                        defaultChecked={wind === 'on'}
                        onClick={onChange}
                    />
                    <Circle />
                </RadioLabel>
                <RadioLabel>
                    Off
                    <RadioButton
                        type="radio"
                        name="wind"
                        value="off"
                        defaultChecked={wind === 'off'}
                        onClick={onChange}
                    />
                    <Circle />
                </RadioLabel>
            </RadioGroup>
        </FieldSet>
    </Form>
);

export default EditorDumb;
