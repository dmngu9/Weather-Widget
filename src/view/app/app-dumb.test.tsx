import * as React from 'react';
import { shallow, mount } from 'enzyme';

import App from './app-dumb';
import { FieldText, RadioButton } from 'view/editor';

describe('<App />', () => {
    it('should render correctly', () => {
        const wrapper = shallow(<App />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should set correct state when title field change', () => {
        const wrapper = mount(<App />);

        wrapper.find(FieldText).simulate('change', { target: { name: 'title', value: 'hello' } });
        wrapper.update();

        expect(wrapper.state('title')).toBe('hello');
    });

    it('should set correct state when unit radio field change', () => {
        const wrapper = mount(<App />);

        wrapper
            .find(RadioButton)
            .filterWhere(radioButton => radioButton.prop('value') === 'imperial')
            .simulate('click', { target: { name: 'unit', value: 'imperial' } });
        wrapper.update();

        expect(wrapper.state('unit')).toBe('imperial');
    });

    it('should set correct state when unit radio field change', () => {
        const wrapper = mount(<App />);

        wrapper
            .find(RadioButton)
            .filterWhere(radioButton => radioButton.prop('value') === 'off')
            .simulate('click', { target: { name: 'unit', value: 'off' } });
        wrapper.update();

        expect(wrapper.state('unit')).toBe('off');
    });
});
