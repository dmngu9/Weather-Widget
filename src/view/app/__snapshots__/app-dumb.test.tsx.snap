// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly 1`] = `
<styled.div>
  <Styled(EditorDumb)
    onChange={[Function]}
    unit="metric"
    wind="on"
  />
  <Styled(Widget)
    showWind={true}
    title="Title of widget"
    unit="metric"
  />
</styled.div>
`;
